datafusion/physical-expr/src/physical_expr.rs

```diff
@@ -128,6 +137,232 @@ impl PhysicalExprStats for BasicExpressionStats {
     }
 }
 
+/// a Trait for marking tree node types that are rewritable
```

Comment:
I recommend putting the rewriting traits into some other module, such as `datafusion/physical-expr/src/rewrite.rs` or example

We can do it as a follow on PR too

Reply:
Sure.  I can change it in this PR also.